ssg/build_yaml.py

```diff
@@ -335,64 +326,141 @@ def __sub__(self, other):
         return profile
 
 
-class ResolvableProfile(Profile):
+class ProfileWithSeparatePolicies(Profile):
     def __init__(self, * args, ** kwargs):
-        super(ResolvableProfile, self).__init__(* args, ** kwargs)
-        self.resolved = False
+        super(ProfileWithSeparatePolicies, self).__init__(* args, ** kwargs)
+        self.policies = {}
+
+    def read_yaml_contents(self, yaml_contents):
+        policies = yaml_contents.pop("policies", None)
+        if policies:
+            self._parse_policies(policies)
+        super(ProfileWithSeparatePolicies, self).read_yaml_contents(yaml_contents)
+
+    def _parse_policies(self, policies_yaml):
+        for item in policies_yaml:
+            id_ = required_key(item, "id")
+            controls = required_key(item, "controls")
+            if not isinstance(controls, list):
+                if controls != "all":
+                    msg = ("Policy {id_} contains invalid controls list {controls}."
+                        .format(id_=id_, controls=str(controls))
+                        )
+                    raise ValueError(msg)
+            self.policies[id_] = controls
 
     def resolve_controls(self, controls_manager):
         for policy_id, controls in self.policies.items():
+            to_expand = Control()
```

Comment:
Can you explain this? Later, it replaces the controls_list which used to be a list of controls objects by a single instance of a control object.

Reply:
I have changed the flow closer to the original one. See the 3rd commit.